comp/core/tagger/impl/local_tagger_mock.go

```diff
@@ -0,0 +1,187 @@
+	taggermock "github.com/DataDog/datadog-agent/comp/core/tagger/mock"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/origindetection"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/tagstore"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/telemetry"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	coretelemetry "github.com/DataDog/datadog-agent/comp/core/telemetry"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	taggertypes "github.com/DataDog/datadog-agent/pkg/tagger/types"
+	"github.com/DataDog/datadog-agent/pkg/tagset"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// FakeTagger is a fake implementation of the tagger interface
+type FakeTagger struct {
+	tagger   tagger.Component
+	tagStore *tagstore.TagStore
+}
+
+// MockRequires is a struct containing the required components for the mock.
+type MockRequires struct {
+	Config       config.Component
+	WorkloadMeta workloadmeta.Component
+	Log          logdef.Component
+	Telemetry    coretelemetry.Component
+}
+
+// MockProvides is a struct containing the mock.
+type MockProvides struct {
+	Comp taggermock.Mock
+}
+
+// NewMock instantiates a new FakeTagger.
+func NewMock(req MockRequires) MockProvides {
+	tagStore := tagstore.NewTagStore(nil)
+	localTagger, err := newLocalTagger(req.Config, req.WorkloadMeta, req.Log, req.Telemetry, tagStore)
+	if err != nil {
+		log.Errorf("Failed to create local tagger: %v", err)
+	}
+
+	return MockProvides{
+		Comp: &FakeTagger{
+			tagger:   localTagger,
+			tagStore: tagStore,
+		},
+	}
+}
+
+// SetTags allows to set tags in tagStore for a given source, entity
+func (f *FakeTagger) SetTags(entityID types.EntityID, source string, low, orch, high, std []string) {
+	f.tagStore.ProcessTagInfo([]*types.TagInfo{
+		{
+			Source:               source,
+			EntityID:             entityID,
+			LowCardTags:          low,
+			OrchestratorCardTags: orch,
+			HighCardTags:         high,
+			StandardTags:         std,
+		},
+	})
+}
+
+// SetGlobalTags allows to set tags in tagStore for the global entity
+func (f *FakeTagger) SetGlobalTags(low, orch, high, std []string) {
+	f.SetTags(types.GetGlobalEntityID(), "static", low, orch, high, std)
+}
+
+// LoadState loads the state for the tagger from the supplied map.
+func (f *FakeTagger) LoadState(state []types.Entity) {
+	if state == nil {
+		return
+	}
+
+	for _, entity := range state {
+		f.tagStore.ProcessTagInfo([]*types.TagInfo{{
+			Source:               "replay",
+			EntityID:             entity.ID,
+			HighCardTags:         entity.HighCardinalityTags,
+			OrchestratorCardTags: entity.OrchestratorCardinalityTags,
+			LowCardTags:          entity.LowCardinalityTags,
+			StandardTags:         entity.StandardTags,
+			ExpiryDate:           time.Time{},
+		}})
+	}
+
+	log.Debugf("Loaded %v elements into tag tagStore", len(state))
+}
+
+// GetTagStore returns the tag store.
+func (f *FakeTagger) GetTagStore() *tagstore.TagStore {
```

Comment:
I think we should try to remove this function from the interface and the implementations.
It exposes an internal struct.

Reply:
This is done as in some tests, we are using `tagStore.ProcessTagInfo()`. Another potential solution would be to allow the `tagStore` to be created before the `FakeTagger` and then passed in the constructor?